spec/controllers/users/invitations_controller_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Users::InvitationsController, type: :controller do
+  let(:user) { create(:user) }
+  let(:token) { JsonWebToken.encode(user_id: user.id) }
+  let(:params) do
+    {
+      user: {
+        first_name: Faker::Name.first_name,
+        last_name: Faker::Name.last_name,
+        phone: Faker::Base.numerify('###-###-####'),
+        email: Faker::Internet.email,
+        avatar: Rack::Test::UploadedFile.new(ENV['LOCAL_IMAGE_PATH']),
+        address: Faker::Address.full_address,
+        member_type: User.member_types.keys.sample,
+        password: 'password'
+      }
+    }
+  end
+
+  before do
+    @request.env['devise.mapping'] = Devise.mappings[:user]
+    request.headers['token'] = token
+    allow(Twilio::TwilioTextMessenger).to receive(:perform)
+  end
+
+  describe 'POST#create' do
+    let(:send_request) { post :create,
+                              params: params.merge(current_user: user),
+                              as: :json }
+
+    subject { response }
+
+    it 'creates user' do
+      expect { send_request }.to change(User, :count).by(1)
+    end
+
+    context 'when invites' do
+      before { send_request }
+
+      it 'returns json' do
+        expect(response.content_type).to include('application/json')
+      end
+      it 'invited user has the same family as inviter' do
+        expect(response.parsed_body['data']['user']['family_id']).
+            to eq(user.family_id)
+      end
+      it { expect(response).to have_http_status(:created) }
+      it_behaves_like 'success action'
+    end
+  end
+
+  describe 'PUT#update' do
+    let!(:user) { create(:user_with_invite) }
+    let(:send_request) { put :update,
+                             params: update_params,
+                             as: :json }
+
+    before { send_request }
+
+    subject { response }
+
+    context 'with correct token' do
+      let(:update_params) do
+        {
+            user: params[:user].merge(invitation_token: "M167myTEewNxU9zMbs-1")
```

Comment:
do not use hardcoded values please

Reply:
i used method `invite!` to generate invitation_token